packages/legend-studio/src/stores/sidebar-state/LocalChangesState.ts

```diff
@@ -293,6 +302,18 @@ export class LocalChangesState {
     if (!localChanges.length) {
       return;
     }
+    yield flowResult(
+      this.sdlcState.fetchWorkspaceHeadRevision(
+        this.sdlcState.activeProject.projectId,
+        this.sdlcState.activeWorkspace,
+      ),
+    );
+    if (this.sdlcState.isActiveRevisionOutdated) {
+      this.editorStore.localChangesState.workspaceRevisionUpdaterState.updateToWorkspaceHeadRevision(
+        true,
+      );
+      return;
```

Comment:
I think we should show a notification here. An actionable notification is even better, e.g. user can click and update their workspace.